### ISSUE TITLE
Warn about discarded constant/pure expressions

### ISSUE THREAD

**Issue by tatut** [clj-kondo/clj-kondo#298]
```
(defn foo []
  1
  2
  3)
```

Has two constant expressions that are discarded (`1` and `2`) and cannot side effect.
Warn about discarded expressions.

Related: often found with a displaced docstring.

**Comment by borkdude** [comment]
Also see #611.

Comment from @jafingerhut about the eastwood var-info.edn file:

> Same warning applies if key :pure-fn has value true, too.  conj! is not a pure function, so I created separate keys for those two properties.

We can probably use that in this issue.

**Comment by borkdude** [comment]
@tatut The above example now works on master.

**Comment by cjsauer** [comment]
This addition is now spewing warnings for all JS interop calls that take strings as arguments. Examples:

```
(js/ajaxRequest "POST" payload callback)
;; Redundant expression: "POST"
```

```
(js/Tipped.create ".query-selector" content opts)
;; Redundant expression: ".query-selector"
```

```
(js/document.implementation.hasFeature "http://www.w3.org/TR/SVG11/feature#Image" "1.1")
;; Redundant expression: "http://www.w3.org/TR/SVG11/feature#Image"
```

It seems to only affect `(js/...)` interop specifically in my project and it is only string arguments that are not in last position that are flagged.

**Comment by cjsauer** [comment]
Ah I found issue https://github.com/clj-kondo/clj-kondo/issues/1185. Linking here for cross reference purposes.
